tests/utils.h

```diff
@@ -126,11 +126,16 @@ struct FitInput
 	std::vector< REAL > initial_parameters;
 	std::vector< int > parameters_to_fit;
 
+	std::vector< REAL > constraints;
+	std::vector< int > constraint_types;
+
 	REAL tolerance;
 	int max_n_iterations;
 
 	std::vector< REAL > user_info_; // user info is REAL
 
+	std::vector< REAL> expected_parameters; // size zero to not check
```

Comment:
this vector holds the true_parameters we set up with, then we compare fit result with this vector, to test for correctness.